proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,100 @@
+# ***Add `nuget list` Functionality to DotNet***
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- [Nigusu](https://github.com/Nigusu-Allehu) <!-- GitHub username link -->
+- [GitHub Issue](https://github.com/NuGet/Home/issues/6060) <!-- GitHub Issue link -->
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+The specification seeks to add `nuget list` functionality to dotnet. The functionality will be added to dotnet as a search command: `dotnet package search`.  
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+This is a customer requested functionality as discussed in this issue: https://github.com/NuGet/Home/issues/6060. Adding the functionality will aid our customers to be able to use the command when writing scripts.
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+Imagine you have your dotnet CLI open. You would like to get the list of all packages from a source `<MySource>`. No worries, you can just use the following command : `dotnet package search --source <MySource>`. It will provide you with the list of all the packages in the source `<MySource>`.
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+The `package search [search terms] [options]` command will have the following options 
+
+| Option | Function |
+|---------|:----------|
+| `--Source` | A list of packages to search |
+| `--ExactMatch` | Return exact matches only as a search result |
+| `--Verbose` | Displays a detailed ist of information for each package |
+| `--Prerelease` | Allow prerelease packages to be shown. |
+| `--IncludeDelisted` | Allow unlisted packages to be shown |
+| `--NonInteractive` | Do not prompt for user input or confirmations.|
+| `--Help` | Show command help and usage information |
+| `--Verbosity` | Display the amount of details in the output: normal, quiet, detailed. |
+| `--ConfigFile` | The NuGet configuration file. If not specified, the hierarchy of configuration files from the current directory will be used. |
```

Comment:
NuGet.exe follows PowerShell conventions, which uses a single `-` and Pascal Case for options. For example `-SomeArg`.

The `dotnet` CLI uses GNU conventions, which is a single `-` for single letter args, and `--` with snake case for multi-letter args. For example, `-s` or `--some-arg`. The single character, single `-` is because GNU's getopt library lets you use `-123` as equivalent to `-1 -2 -3`. I don't know if System.CommandLine, and therefore the `dotnet` CLI supports this.

These option names should be updated to meet the appropriate naming convention.

Reply:
This GNU behavior of combining single-letter flags is called 'bundling' and System.CommandLine (along with the dotnet CLI) do support this.